docs/plugin_author_guide/compiler_plugins.rst

```diff
@@ -0,0 +1,237 @@
+.. _compiler_plugins:
+
+Compiler Plugins
+================
+
+Most concrete algorithms in Metagraph are self-contained "black box
+implementations" that are run independently by the dispatcher or Dask
+scheduler.  However, Metagraph also allows a special kind of concrete
+algorithm to be defined which is "compilable."  A compilable algorithm is
+described in such a way that a compiler plugin can do just-in-time (JIT)
+compilation of the algorithm to a fast, executable form (usually machine
+code).  The benefit of JIT compilation is that multiple tasks can be fused
+together by the compiler to reduce redundant work and eliminate the Python
+call overhead associated with each individual task.
+
+.. note::
+
+    In the future, JIT compilation may also allow specialization of algorithms
+    based on input constants.
+
+When using Metagraph with Dask (see :ref:`dask`), a special Dask graph
+optimizer is invoked when the ``compute()`` method is called on a
+``Placeholder`` object.  This optimizer will scan the Dask task graph for
+compilable Metagraph algorithms and identify task subgraphs to send to the
+compiler plugin.  Currently, subgraphs must be linear chains of compilable
```

Comment:
This is the first mention of the term "linear chain". For extra clarity, it might be a good idea to have a visual showing what a linear chain is, e.g. reusing `dag_compile_subgraphs.png` here might be a good idea.

Reply:
linking to the dask documentation, which has a good general discussion of how this all works (with pictures)